matter_server/rootfs/etc/s6-overlay/s6-rc.d/matter-server/run

```diff
@@ -30,7 +30,17 @@ if ! bashio::var.has_value "${server_port}"; then
     extra_args+=('--listen-address' "$(bashio::addon.ip_address)")
 fi
 
-primary_interface="$(bashio::api.supervisor 'GET' '/network/info' '' 'first(.interfaces[] | select (.primary == true)) .interface')"
+if ! bashio::config.exists primary_interface || [ -z $(bashio::config primary_interface) ]; then
+  bashio::log.info 'No primary_interface set in config, fallback to using supervisor primary interface'
+  primary_interface="$(bashio::api.supervisor 'GET' '/network/info' '' 'first(.interfaces[] | select (.primary == true)) .interface')"
+else
+  primary_interface=$(bashio::string.lower "$(bashio::config primary_interface)")
+  bashio::log.info "primary_interface set in config to ${primary_interface}"
+  if ! ip ad ls dev ${primary_interface} > /dev/null ; then
+    bashio::log.warning 'Specified interface does not exist. Falling back to supervisor primary interface'
+    primary_interface="$(bashio::api.supervisor 'GET' '/network/info' '' 'first(.interfaces[] | select (.primary == true)) .interface')"
+  fi
+fi
```

Comment:
This was pervious functionality that I didn't write. I'd rather not change this, since this could break some previously working cases. The downstream software may have additional error recovery for this case that exiting here could break.